dd-sdk-android-compose/src/main/kotlin/com/datadog/android/compose/InteractionTracking.kt

```diff
@@ -55,3 +158,273 @@ internal class TapActionTracker(
         onTap.value.invoke()
     }
 }
+
+@Suppress("LongParameterList")
+internal suspend fun trackSwipe(
+    rumMonitor: RumMonitor,
+    targetName: String,
+    interactionSource: InteractionSource,
+    interactionType: InteractionType.Swipe<*>,
+    isRtl: Boolean,
+    attributes: Map<String, Any?>
+) {
+    trackDragInteraction<SwipeStartProps>(
+        interactionSource,
+        onStart = { interactions, start ->
+            interactions[start] = SwipeStartProps(
```

Comment:
Hmm I would rather put this in the `trackDragInteraction` function where you call this lambda than here ? And just keep that RUM call here, that's makes it more visible don't you think ?

Reply:
Start properties objects are different for swipe and scroll interaction, so we have to keep it here.

At the beginning I was trying to do indeed everything in the single `trackInteraction` function, but that led to the many `is` checks and it wasn't easy to follow logically. So I decided to split it and extract common type-independent logic.